src/rviz/default_plugin/odometry_display.cpp

```diff
@@ -372,6 +368,23 @@ void OdometryDisplay::update(float /*wall_dt*/, float /*ros_dt*/)
 
   assert(arrows_.size() == axes_.size());
   assert(axes_.size() == covariance_property_->sizeVisual());
+
+  // applyTransforms
+  if(!last_used_message_)
+    return;
+
+  Ogre::Vector3 position;
+  Ogre::Quaternion orientation;
+  if (!context_->getFrameManager()->getTransform(last_used_message_->header.frame_id, ros::Time(),
+                                                 position, orientation))
+  {
+    ROS_ERROR("Error transforming odometry '%s' from frame '%s' to frame '%s'", qPrintable(getName()),
+              last_used_message_->header.frame_id.c_str(), qPrintable(fixed_frame_));
```

Comment:
Even though the left/old one looks better, the right one is correct. Our dead reckoning (red) is really that bad ;) It is only based on wheel turn rates and a yaw gyroscope from the ESP sensor. Therefore it only works in the 2D plane as it cannot detect roll and pitch of the vehicle. The inertial navigation system, on the other hand, is able to measure roll and pitch. Thats why the blue trajectory pitches and rolls w.r.t. to the fixed frame (odom, which is based on our dead reckoning).

That the left one looks meaningful is a coincidence because the trajectories are "similar". New Odometry messages, or more precisely the arrows, are inserted at the right position of the scene but old ones are not updated when the fixed frame moves w.r.t. these arrows.

Reply:
> If you are at it: Could you replace this console error output with a display status error (`Display::setStatus`)?
> Remember to reset the status to OK if there is no error.

I added this in the latest commit. I borrowed the code from `grid_display.cpp`.

Reply:
I finally removed the setStatus commit again, as this part is already handled by the MessageFilterDisplay Super-Class. Otherwise there would be dup
![Screenshot from 2021-06-22 20-56-12](https://user-images.githubusercontent.com/11663218/122985930-09a4fb00-d39f-11eb-8cd7-baf737da6e5c.png)
licate Transform items:

Reply:
Sure. Sorry for the noise.